### PR TITLE
Fix issue with incorrectly finding and triggering certain remappings

### LINKED ISSUE #2888: 'r' in insert mode not entered when typed quickly
Issue Type: <b>Bug</b>

Was originially filed against VS Code by @wcauchois. Pls see: https://github.com/Microsoft/vscode/issues/54959.

Extension version: 0.15.6
VS Code version: Code - Insiders 1.26.0-insider (befea2253626b1918352df04477cae83eee618ac, 2018-07-24T05:11:44.849Z)
OS version: Darwin x64 16.7.0

### PR DESCRIPTION
**Pull request by xconverge**
fixes #2888

### CODE REVIEW

**Comment by wcauchois** [comment]
Thank you @xconverge!

**Comment by jpoon** [comment]
Err. sorry. I don't get this change. ELI5?

**Comment by xconverge** [comment]
@jpoon 

Running this:

```
var remappings: { [key: string]: number } = {};

remappings["test"] = 4;

if(remappings["constructor"]){
	console.log("matched constructor");
}

if(remappings["test"]){
	console.log("matched matched test");
}

if(remappings["test123"]){
	console.log("matched matched test123");
}
```

The output is:

```
console: matched constructor
console: matched matched test
```

Any function name in the prototype of the object counts as a key in the map.

The fix here was to remove the prototype and have a plain map.

This was super enlightening to me and does not make sense to me coming from C++

**Comment by jpoon** [comment]
Lol. Good catch
